extents/report_ftk_extents.py

```diff
@@ -0,0 +1,271 @@
+    The function returns the entire list.
+    '''
+
+    tree = tree.xpath(
+        '/fo:root/fo:page-sequence[@master-reference="TOC"]/fo:flow',
+        namespaces=FO_NAMESPACE
+    )[0]
+
+    ers = []
+    hierarchy = []
+    for child in tree:
+        # skip rows with an indent < 24
+        if not child.get("start-indent"):
+            continue
+
+        indent = int(child.get("start-indent").split(sep="pt")[0])
+        level = (indent//12) - 2
+
+        if level >= 0:
+            # build a list of parents based on level
+            if level <= len(hierarchy) - 1:
+                hierarchy = hierarchy[:level]
+            elif level > len(hierarchy) + 1:
+                raise ValueError(
+                    f'Unexpected jump in hierarchy at {child.text}'
+                )
+            hierarchy.append(child.text)
+
+            # only record if entry is an ER
+            possible_ref = child.xpath(
+                'fo:basic-link/fo:page-number-citation', namespaces=FO_NAMESPACE
+            )
+            if possible_ref and hierarchy[-1].startswith('ER'):
+                refid = possible_ref[0].get('ref-id')
+                ers.append(
+                    ['/'.join(hierarchy.copy()), refid]
+                )
+
+    return ers
+
+
+def transform_bookmark_tables(
+    tree: etree.ElementTree
+) -> list[list[str, str, str]]:
+
+    '''
+    transforms each row in the 'bookmarksPage' table
+    into a string. this string contains all the extent information
+    that will be summarized later.
+    the return is a list of lists where the first item is the id with
+    the prefix bk and the second item is a string serialized from the XML.
+    '''
+
+    extent_tree = tree.xpath(
+        '/fo:root/fo:page-sequence[@master-reference="bookmarksPage"]/fo:flow/fo:table[@id]'\
+        ,
+        namespaces=FO_NAMESPACE
+    )
+
+    bookmark_contents = []
+    for row in extent_tree:
+
+        #row is an /fo:row in /fo:table[@id]
+
+        bookmark_id = row.get('id')
+        id_number = bookmark_id.split('_')[0]
+        file_table = etree.tostring(row, method='text', encoding="UTF-8")
+        bookmark_contents.append([bookmark_id, id_number, file_table])
+
+    return bookmark_contents
+
+
+def add_extents_to_ers(
+    er_list: list[list[str, str]],
+    bookmark_tables: list[list[str, int, int]]
+) -> list[list[str, int, int]]:
+
+    '''
+    summarizes the extent for each ER by
+    correlating the table of contents with the bookmark tables.
+    Returns list of lists with hierarchal ER string, file size, and file count.
+    '''
+
+    ers_with_extents = []
+
+    for er in er_list:
+        bookmark_id = er[1]
+        size, count = get_er_report(bookmark_tables, bookmark_id)
+        ers_with_extents.append([er[0], size, count])
+
+    return ers_with_extents
+
+
+def get_er_report(
+    er_files: list[str, str, str],
+    bookmark_id: str
+) -> tuple(int, int):
```

Comment:
tuple appears to want a list passed to it "tuple([int,int])" otherwise you get a TypeError "TypeError: tuple expected at most 1 argument, got 2"

Reply:
I fixed this with 98359c2